File: 70_ClimbingStairs.java
class Solution {
    public int climbStairs(int n) {
        if(n == 1) return 1;
    	int a = 1, b = 2;
        for(int i = 3; i <= n; i++) {
        	int c = b;
        	b = a + b;
        	a = c;
        }
        return b;
    }
}

File: c++/70_ClimbingStairs.cpp
class Solution {
public:
    int numWays(int n) {
        if (n <= 2) return n;
        int a = 0, b = 0, c = 1;
        for (int i = 1; i <= n; i++){
            a = b;
            b = c;
            c = a + b;
        }
        return c;
    }
};
